tests/integration/test_authd/test_authd_force_options.py

```diff
@@ -0,0 +1,252 @@
+authd_default_password_path = os.path.join(WAZUH_PATH, 'etc', 'authd.pass')
+force_options_tests = load_tests(os.path.join(test_data_path, 'force_options.yaml'))
+
+DEFAULT_FORCE_INSERT = 'yes'
+DEFAULT_USE_USER_IP = 'no'
+CLIENT_KEY_ENTRY_LEN = 4
+
+conf_params = {'USE_SOURCE_IP': [], 'FORCE_INSERT': []}
+
+for case in force_options_tests:
+    conf_params['USE_SOURCE_IP'].append(case.get('USE_SOURCE_IP', DEFAULT_USE_USER_IP))
+    conf_params['FORCE_INSERT'].append(case.get('FORCE_INSERT', DEFAULT_FORCE_INSERT))
+
+p, m = generate_params(extra_params=conf_params, modes=['scheduled'] * len(force_options_tests))
+
+configurations = load_wazuh_configurations(configurations_path, __name__, params=p, metadata=m)
+
+# Variables
+log_monitor_paths = []
+receiver_sockets_params = [(("localhost", 1515), 'AF_INET', 'SSL_TLSv1_2')]
+monitored_sockets_params = [('wazuh-modulesd', None, True), ('wazuh-db', None, True), ('wazuh-authd', None, True)]
+receiver_sockets, monitored_sockets, log_monitors = None, None, None  # Set in the fixtures
+# fixtures
+
+test_index = 0
+
+
+def get_current_test():
+    global test_index
+    current = test_index
+    test_index += 1
+    return current
+
+
+@pytest.fixture(scope="module", params=configurations)
+def get_configuration(request):
+    """Get configurations from the module"""
+    return request.param
+
+
+def clean_client_keys_file():
+    try:
+        client_file = open(client_keys_path, 'w')
+        client_file.close()
+    except IOError as exception:
+        raise
+
+
+def read_hostname():
+    return socket.gethostname()
+
+
+def override_wazuh_conf(configuration):
+    # Stop Wazuh
+    control_service('stop', daemon='wazuh-authd')
+    time.sleep(1)
+    check_daemon_status(running=False, daemon='wazuh-authd')
+    truncate_file(LOG_FILE_PATH)
+
+    # Configuration for testing
+    test_config = set_section_wazuh_conf(configuration.get('sections'))
+    # Set new configuration
+    write_wazuh_conf(test_config)
+
+    # reset_client_keys
+    clean_client_keys_file()
+
+    time.sleep(1)
+    # Start Wazuh
+    control_service('start', daemon='wazuh-authd')
+
+    """Wait until agentd has begun"""
+
+    def callback_agentd_startup(line):
+        if 'Accepting connections on port 1515' in line:
+            return line
+        return None
+
+    log_monitor = FileMonitor(LOG_FILE_PATH)
+    log_monitor.start(timeout=30, callback=callback_agentd_startup)
+    time.sleep(1)
+
+
+def check_client_keys_file(response):
+    try:
+        with open(client_keys_path) as client_file:
+            client_lines = client_file.readlines()
+            for line in client_lines:
+                # discard endline because response finalize with "'\n\n" character and client.keys only with "\n"
+                if line[:-1] in response:
+                    client_file.close()
+                    return True
+    except IOError as exception:
+        raise
```

Comment:
Why catch an exception to raise it?
```suggestion
    with open(client_keys_path) as client_file:
       client_lines = client_file.readlines()
        for line in client_lines:
        # discard endline because response finalize with "'\n\n" character and client.keys only with "\n"
            if line[:-1] in response:
               return True
```